src/Hl7.Fhir.Base/Serialization/JsonSerializerOptionsExtensions.cs

```diff
@@ -96,6 +98,54 @@ public static JsonSerializerOptions ForFhir(this JsonSerializerOptions options,
 
             return options;
         }
+        
+        /// <summary>
+        /// Modify the options to use a preset list of errors to ignore by specifying a mode. This can be any member of <see cref="DeserializerModes"/>
+        /// </summary>
+        /// <remarks>
+        /// Modifying the options is always left-associative. This means that defining custom constraints should probably be done AFTER setting the mode.
+        /// </remarks>
+        public static JsonSerializerOptions UsingMode(this JsonSerializerOptions options, DeserializerModes mode)
+        {
+            var factory = getCustomFactoryFromList(options.Converters);
+            
+            factory.IgnoreFilter = mode switch
+            {
+                DeserializerModes.Recoverable => FilterPredicateExtensions.IsRecoverableIssue,
```

Comment:
This is somewhat of a duplicate of the factory methods in the `FhirSerializationEngineFactory`. In fact, there is a risk that the mode switch is different from creating a specific engine. In fact, setting the mode to Ostrich would only take into account the list of errors, while it is also important that the validators are set to null (for performance reasons). I know you deleted that code, but it is relevant.  So in future updates we run the risk of updating the factory methods and not this or vice versa. Can we find a smart solution to this?

Reply:
Maybe, when setting deserializer modes, we shouldn't swap filters, but we should swap engines -> create the engine with the factory and then set it in the CustomFactory?

Reply:
We want the predicates to be called on the (known) base predicate. I had the implementation like this at first, passing on the engine mode, but this did not work, since the intended behaviour for .Enforcing() has to be able to bind to something (some base mode) to keep track of the .ignoring and .enforcing combined predicate